### ISSUE TITLE
Failed to decode the image below:

### ISSUE THREAD

**Issue by dmitryvm** [kaksmet/jpeg-decoder#96]
[http://www.filedropper.com/test_17](http://www.filedropper.com/test_17)
```
thread 'main' panicked at 'failed to decode image: Format("no data found")', src/libcore/result.rs:997:5
```

**Comment by kaksmet** [comment]
This is the same bug as described in image-rs/image#738

**Comment by kaj** [comment]
I can't seem to get an image from the filedropper link.  Did it expire?

**Comment by alu** [comment]
@kaksmet 
Hi.

I encountered `spectral selection is not allowed in non-progressive scan` Error.
`convert` command in Linux displayed `Invalid SOS parameters for sequential JPEG` warning but the image was generated successfully.
Can i get same behavior in this crate?

**Comment by alu** [comment]
When I looked at the Exif of the image in question, it was taken with the following hardware.

samsung SCV36
samsung SC-02G
samsung SC-05G

We handle a myriad of jpeg files, and some of them have been confirmed.

**Comment by alu** [comment]
This question is may be relevant

https://stackoverflow.com/questions/48653584/cannot-load-jpeg-with-java-created-by-samsung-phone

**Comment by alu** [comment]
@fintelia 
Do you have any plans to address this issue in the future?
I don't have any skills for jpeg encoding so all i can do is provide information, sorry.

**Comment by lovasoa** [comment]
@alu : I don't think this is the same issue. The issue here is about the `no data found` error generated when a progressive JPEG image lacks a final scan. Maybe you should open another issue.

**Comment by alu** [comment]
@lovasoa 

Thank you for your reply.
I created [new ussie](https://github.com/image-rs/jpeg-decoder/issues/130)

**Comment by Shnatsel** [comment]
More sample issues triggering this behavior can be found in https://github.com/image-rs/image/issues/1234

**Comment by Shnatsel** [comment]
FWIW this is the only error triggered on the [archive.org wallhaven dataset](https://archive.org/details/WallhavenSexyWomen), which is the largest easily downloadable dataset of real-world images I could find (100,000 files, 76Gb).

There are also machine learning datasets, but those tend to be heavily preprocessed so they don't show much variance in encoding options.
